PacketDefinitions420/PacketHandlerManager.cs

```diff
@@ -275,21 +285,37 @@ public bool HandlePacket(Peer peer, ENet.Packet packet, Channel channelId)
             // every packet that is not blowfish go here
             if (data.Length >= 8)
             {
-                // TODO: each user will have his unique key
-                data = _blowfish.Decrypt(data);
+                ulong playerId = _peers.First(x => x.Value.Address.Equals(peer.Address)).Key;
+                data = _playerBlowFishKeys[playerId].Decrypt(data);
             }
             return HandlePacket(peer, data, channelId);
         }
 
         private bool HandleHandshake(Peer peer, byte[] data)
         {
             var request = PacketReader.ReadKeyCheckRequest(data);
-            // TODO: keys for every player
-            ulong playerID = (ulong)_blowfish.Decrypt(request.CheckId);
 
-            // Wrong Blowfish key, or the client is already connected
-            if (request.PlayerID != playerID || _peers.ContainsKey(request.PlayerID))
+            bool clientFound = false;
+            foreach(var blowfish in _blowfishes)
+            {
+                ulong playerID = (ulong)blowfish.Decrypt(request.CheckId);
+                if(request.PlayerID == playerID)
+                {
+                    if(_peers.ContainsKey(request.PlayerID) || _playerBlowFishKeys.ContainsValue(blowfish))
```

Comment:
Why `_playerBlowFishKeys.ContainsValue(blowfish)`? It can be that two players chose the same key. The main problem is if users try to connect with the id of other player.

Reply:
OK. I see that that in this case it still works. Though I still don't understand why this check is necessary.

Reply:
This check is indeed uneccessary, I thought playerId relates to something else for some reason. Will fix that.